pkg/cmd/roachtest/tests/costfuzz.go

```diff
@@ -35,6 +36,13 @@ func registerCostFuzz(r registry.Registry) {
 		case sqlsmith.SeedMultiRegionSetupName:
 			clusterSpec = r.MakeClusterSpec(9, spec.Geo(), spec.GatherCores())
 		case "workload-replay":
+			// When running in CI, only allow running in the private roachtest, which
+			// has the required credentials.
+			if os.Getenv("TC_BUILD_ID") != "" && os.Getenv("ROACHTEST_PRIVATE") != "1" {
+				fmt.Println("Skipping costfuzz/workload-replay in public CI roachtest")
+				continue
```

Comment:
Setting a `Skip` reason instead of not registering the test at all is probably friendlier to folks looking at the list of tests in the nightly build.

Reply:
I thought about that but then it would try and allocate a cluster before skipping - though it's probably better for consistency. Updating.

Reply:
I meant setting the `Skip` field on the `TestSpec` struct, in which case the cluster won't be allocated.